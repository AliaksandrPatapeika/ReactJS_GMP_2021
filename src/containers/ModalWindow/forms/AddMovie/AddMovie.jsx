import '../../ModalWindow.less';

import PropTypes from 'prop-types';
import React, {useContext, useState} from 'react';

import Button from '../../../../components/Button';
import Select from '../../../../components/Select';
import {Genres} from '../../../../constants';
import MovieContext from '../../../../context';
import {MockOnClick} from '../../../../mocks/mockData';

const AddMovie = ({formTitle}) => {
  const [optionSelected, setOptionSelected] = useState(null);

  const selectChange = (selected) => {
    setOptionSelected(selected);
  };

  const selectReset = () => {
    setOptionSelected(null);
  };

  const {activeModalWindow} = useContext(MovieContext);

  return (
    <>
      {activeModalWindow === 'addMovie' && (
      <>
        <span className="title">{formTitle}</span>
        <form onSubmit={() => MockOnClick('form Add movie onSubmit')} onReset={selectReset}>
          <div className="formInput">
            <label htmlFor="title" className="label">
              TITLE
            </label>
            <input
              type="text"
              name="title"
              id="title"
              className="input"
              placeholder="Title here"
              defaultValue="Moana"
              autoComplete="off"
            />
          </div>
          <div className="formInput">
            <label htmlFor="releaseDate" className="label">
              RELEASE DATE
            </label>
            <input
              type="date"
              name="releaseDate"
              id="releaseDate"
              className="input"
              placeholder="Select Date"
              required
            />
          </div>
          <div className="formInput">
            <label htmlFor="movieURL" className="label">
              MOVIE URL
            </label>
            <input
              type="text"
              name="movieURL"
              id="movieURL"
              className="input"
              placeholder="Movie URL here"
              autoComplete="off"
            />
          </div>
          <div className="formInput">
            <label htmlFor="genres" className="label">
              GENRE
            </label>
            <Select
              className="select"
              classNamePrefix="select"
              id="genres"
              options={Genres}
              placeholder="Select Genre"
              isMulti
              closeMenuOnSelect={false}
              hideSelectedOptions={false}
              onChange={selectChange}
              allowSelectAll
              allSelectedLabel="All genres"
              value={optionSelected}
            />
          </div>
          <div className="formInput">
            <label htmlFor="overview" className="label">
              OVERVIEW
            </label>
            <input
              type="text"
              name="overview"
              id="overview"
              className="input"
              placeholder="Overview here"
              autoComplete="off"
            />
          </div>
          <div className="formInput">
            <label htmlFor="runtime" className="label">
              RUNTIME
            </label>
            <input
              type="text"
              name="runtime"
              id="runtime"
              className="input"
              placeholder="Runtime here"
              autoComplete="off"
            />
          </div>
          <div className="modalFooter">
            <Button
              type="reset"
              className="button"
              title="RESET"
            />
            <Button
              type="submit"
              className="button"
              title="SUBMIT"
            />
          </div>
        </form>
      </>
      )}
    </>
  );
};

AddMovie.propTypes = {
  formTitle: PropTypes.string.isRequired
};

export default AddMovie;
