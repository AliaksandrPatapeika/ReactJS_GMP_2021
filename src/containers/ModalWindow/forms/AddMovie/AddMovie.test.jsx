import React from 'react';
import renderer from 'react-test-renderer';

import MovieContext from '../../../../context';
import AddMovie from './AddMovie';

describe('AddMovie', () => {
  test('renders correctly', () => {
    const addMovie = renderer
      .create(
        <MovieContext.Provider value={{
				  activeModalWindow: 'addMovie'
        }}
        >
          <AddMovie formTitle="title" />
        </MovieContext.Provider>
      )
      .toJSON();

    expect(addMovie).toMatchSnapshot();
  });
});
