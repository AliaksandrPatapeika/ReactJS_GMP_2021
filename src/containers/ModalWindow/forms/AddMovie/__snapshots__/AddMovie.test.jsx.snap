// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovie renders correctly 1`] = `
Array [
  <span
    className="title"
  >
    title
  </span>,
  <form
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="title"
      >
        TITLE
      </label>
      <input
        autoComplete="off"
        className="input"
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="Title here"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="release_date"
      >
        RELEASE DATE
      </label>
      <input
        className="input"
        id="release_date"
        name="release_date"
        onChange={[Function]}
        placeholder="Select Date"
        required={true}
        type="date"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="poster_path"
      >
        MOVIE URL
      </label>
      <input
        autoComplete="off"
        className="input"
        id="poster_path"
        name="poster_path"
        onChange={[Function]}
        placeholder="Movie URL here"
        type="url"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="genres"
      >
        GENRE
      </label>
      <div
        className="requiredSelectContainer"
      >
        <div
          className="select css-2b097c-container"
          id="genres"
          onKeyDown={[Function]}
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className="select__control css-yk16xz-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="select__value-container select__value-container--is-multi css-g1d714-ValueContainer"
            >
              <div
                className="select__placeholder css-1wa3eu0-placeholder"
              >
                Select Genre
              </div>
              <div
                className="css-b8ldur-Input"
              >
                <div
                  className="select__input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-2-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className="select__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className="select__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <i
                  aria-hidden="true"
                  className="fa fa-caret-down"
                />
              </div>
            </div>
          </div>
        </div>
        <input
          autoComplete="off"
          onChange={[Function]}
          required={true}
          style={
            Object {
              "height": 0,
              "opacity": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
          tabIndex={-1}
          value={Array []}
        />
      </div>
    </div>
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="overview"
      >
        OVERVIEW
      </label>
      <input
        autoComplete="off"
        className="input"
        id="overview"
        name="overview"
        onChange={[Function]}
        placeholder="Overview here"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="runtime"
      >
        RUNTIME
      </label>
      <input
        autoComplete="off"
        className="input"
        id="runtime"
        name="runtime"
        onChange={[Function]}
        placeholder="Runtime here"
        type="number"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="budget"
      >
        BUDGET
      </label>
      <input
        autoComplete="off"
        className="input"
        id="budget"
        name="budget"
        onChange={[Function]}
        placeholder="Budget here"
        type="number"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="revenue"
      >
        REVENUE
      </label>
      <input
        autoComplete="off"
        className="input"
        id="revenue"
        name="revenue"
        onChange={[Function]}
        placeholder="Revenue here"
        type="number"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label required-label"
        htmlFor="tagline"
      >
        TAGLINE
      </label>
      <input
        autoComplete="off"
        className="input"
        id="tagline"
        name="tagline"
        onChange={[Function]}
        placeholder="Tagline here"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="vote_average"
      >
        VOTE AVERAGE
      </label>
      <input
        autoComplete="off"
        className="input"
        id="vote_average"
        name="vote_average"
        onChange={[Function]}
        placeholder="Vote average here"
        step="0.1"
        type="number"
      />
    </div>
    <div
      className="formInput"
    >
      <label
        className="label"
        htmlFor="vote_count"
      >
        VOTE COUNT
      </label>
      <input
        autoComplete="off"
        className="input"
        id="vote_count"
        name="vote_count"
        onChange={[Function]}
        placeholder="Vote count here"
        type="number"
      />
    </div>
    <div
      className="modalFooter"
    >
      <button
        className="button"
        disabled=""
        id=""
        onClick={[Function]}
        type="reset"
      >
        RESET
        
      </button>
      <button
        className="button"
        disabled=""
        id=""
        onClick={[Function]}
        type="submit"
      >
        SUBMIT
        
      </button>
    </div>
  </form>,
]
`;
